app/Models/Feed.php

```diff
@@ -489,6 +542,95 @@ public function loadEntries(SimplePie $simplePie) {
 		}
 	}
 
+	/**
+	 * @param array<string,mixed> $attributes
+	 * @return SimplePie|null
+	 */
+	public function loadHtmlXpath(bool $loadDetails = false, bool $noCache = false, array $attributes = []) {
+		if ($this->url == '') {
+			return null;
+		}
+		$feedSourceUrl = htmlspecialchars_decode($this->url, ENT_QUOTES);
+		if ($this->httpAuth != '') {
+			$feedSourceUrl = preg_replace('#((.+)://)(.+)#', '${1}' . $this->httpAuth . '@${3}', $feedSourceUrl);
+		}
```

Comment:
Cf. discussion in https://github.com/FreshRSS/FreshRSS/pull/4220#issuecomment-1050289052

I don't have time to check right now but there might be something double going on? I think the XPath output is already something akin to `htmlspecialchars_decode()` out of the box, unless I'm mistaken.

Reply:
@math-GH Btw, what's a sample page? I don't really want to create a testcase myself. ;-)

Reply:
see my next comment

Reply:
https://github.com/FreshRSS/FreshRSS/pull/4220/commits/998babbe94755c832a226c79527abbd5b466bbe3
The rest looks good